source/extensions/filters/listener/proxy_protocol/proxy_protocol.cc

```diff
@@ -443,6 +456,24 @@ ReadOrParseState Filter::readProxyHeader(Network::ListenerFilterBuffer& buffer)
   } else {
     // continue searching buffer from where we left off
     for (; search_index_ < raw_slice.len_; search_index_++) {
+
+      if (config_.get()->allowRequestsWithoutProxyProtocol()) {
+        if (search_index_ < PROXY_PROTO_V1_SIGNATURE_LEN &&
+            buf[search_index_] != PROXY_PROTO_V1_SIGNATURE[search_index_]) {
+          possibly_v1_ = false;
+        }
+        if (search_index_ < PROXY_PROTO_V2_SIGNATURE_LEN &&
+            buf[search_index_] != PROXY_PROTO_V2_SIGNATURE[search_index_]) {
+          possibly_v2_ = false;
+        }
+        if (!possibly_v1_ && !possibly_v2_) {
+          // the bytes we have seen so far do not match v1 or v2 proxy protocol, so we can safely
+          // short-circuit
+          ENVOY_LOG(debug, "request does not use v1 or v2 proxy protocol, forwarding as is");
+          return ReadOrParseState::SkipFilter;
+        }
+      }
+
       if (buf[search_index_] == '\n' && buf[search_index_ - 1] == '\r') {
```

Comment:
Should we do the above check in the `else` branch? like:

```
if (buf[search_index_] == '\n' && buf[search_index_ - 1] == '\r') {
....
} else if (config_.get()->allowRequestsWithoutProxyProtocol()) {
....
}
```

Reply:
Addressed in https://github.com/envoyproxy/envoy/pull/18951/commits/a0c977f6ed37cea7349d611cf97a687921c62bd2